packages/rule-components/src/RuleTable/RuleTable.js

```diff
@@ -0,0 +1,249 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Table, TableBody, TableHeader, sortable } from '@patternfly/react-table';
+import { Skeleton } from '@redhat-cloud-services/frontend-components/components/Skeleton';
+import { TableToolbar } from '@redhat-cloud-services/frontend-components/components/TableToolbar';
+import { Pagination, PaginationVariant } from '@patternfly/react-core/dist/js/components/Pagination';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/components/PrimaryToolbar';
+import flatten from 'lodash/flatten';
+import debounce from 'lodash/debounce';
+import { calculateMeta, calculateActiveFilters, createRows } from './helpers';
+
+class RuleTable extends Component {
+    state = {
+        filterValues: {},
+        expanded: []
+    }
+
+    debounceFetchData = debounce(options => {
+        const { fetchData } = this.props;
+        fetchData(options);
+    }, 500)
+
+    componentDidUpdate(prevProps) {
+        if (JSON.stringify(this.props.filterValues) !== JSON.stringify(prevProps.filterValues)) {
+            this.setState({
+                ...this.state,
```

Comment:
I'm not sure I think this line is useless, because setState automatically merge your new object and state object.

Reply:
Yeah, I had a bit more complicated state setter here so I kept the spread in here, but it is better to update just the filterValues.